### PR TITLE
Added lodash intersection and intersectionWith to CollectionUtils

### PR DESCRIPTION
**Pull request by dtnhinh**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

**Review by jstevenson — APPROVED** [review]

File: src/utilities/collection-utils.test.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ describe("CollectionUtils", () => {
             const result = CollectionUtils.hasValues(collection);
 
             // Assert
-            expect(result).toBeTrue();
+            expect(result).toBe(true);
```
Comment (wintondeshong) [inline src/utilities/collection-utils.test.ts:19]: Question: Why the change from the `toBeTrue` matcher?

**Review by wintondeshong — CHANGES_REQUESTED** [review]
@nhinhdao The addition of `intersection` and `intersectionBy` are good by me. I am curious about the change to the matches below. Just let me know

File: src/utilities/collection-utils.test.ts (hunk 1 above)
Comment (brandongregoryscott) [reply src/utilities/collection-utils.test.ts:19]: If I had to guess, it's probably an IDE thing - I believe we still need to update `setupTests.ts` to have an ES6-style import of `jest-extended` for the TS server to properly recognize them in an editor like VS Code. However, I would suggest changing them back, @dtnhinh. We can open an issue to address the `jest-extended` import

File: src/utilities/collection-utils.ts [hunk 2]
```diff
@@ -196,6 +196,8 @@ export const CollectionUtils = {
     hasValues: _hasValues,
     isEmpty: _isEmpty,
     isNotEmpty: _isNotEmpty,
+    intersection: _.intersection,
+    intersectionWith: _.intersectionWith,
```
Comment (brandongregoryscott) [inline src/utilities/collection-utils.ts:200]: Based on #69, should we consider porting these over to be wrapped in a private method instead? @wintondeshong  / @KevinBusch I assume this would just be something short like `const _intersection = (...lodash function signature) => _.intersection(args);`

**Review by brandongregoryscott — CHANGES_REQUESTED** [review]

File: src/utilities/collection-utils.ts (hunk 2 above)
Comment (KevinBusch) [reply src/utilities/collection-utils.ts:200]: Yes, a private function call would be preferred.  It really only makes a difference though if we refactor all the current lodash references though

Comment (brandongregoryscott) [reply src/utilities/collection-utils.ts:200]: Gotta start somewhere 😛

**Review by brandongregoryscott — APPROVED** [review]
Thanks for updating all of those lodash references into private functions, and updating the jest matchers 👍

**Review by wintondeshong — APPROVED** [review]
Thanks for adding this @dtnhinh !

**Comment by brandongregoryscott** [comment]
@dtnhinh Just a reminder to link the related issue in the description with [closing keywords](https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords) next time 👍 Thanks for the contribution!
